sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1759,13 +1759,73 @@ def _df_getitem_unicode_idx_impl(self, idx):
     ty_checker.raise_exc(idx, expected_types, 'idx')
 
 
+def df_getitem_tuple_at_codegen(self, row, col):
+    """
+    Example of generated implementation:
+        def _df_getitem_tuple_at_impl(self, idx):
+            row, _ = idx
+            check_row = False
+            for i in prange(len(self._dataframe.index)):
+                if self._dataframe.index[i] == row:
+                    check_row = True
+            if check_row:
+                data = self._dataframe._data[2]
+                res_data = pandas.Series(data, index=self._dataframe.index)
+                return res_data.at[row]
+            raise IndexingError('Index is out of bounds for axis')
+    """
+    func_lines = ['def _df_getitem_tuple_at_impl(self, idx):']
+    for i in range(len(self.columns)):
+        if self.columns[i] == col:
+            func_lines += [
+                '  row, _ = idx',
+                '  check_row = False',
+                '  for i in prange(len(self._dataframe.index)):',
+                '    if self._dataframe.index[i] == row:',
+                '      check_row = True',
+                '  if check_row:',
+                f'    data = self._dataframe._data[{i}]',
+                '    res_data = pandas.Series(data, index=self._dataframe.index)',
+                '    return res_data.at[row]',
+                '  raise IndexingError("Index is out of bounds for axis")'
+            ]
+            break
+    else:
+        raise IndexingError('Index is out of bounds for axis')
+
+    func_text = '\n'.join(func_lines)
+
+    global_vars = {'pandas': pandas,
+                   'prange': prange,
+                   'IndexingError': IndexingError}
+
+    return func_text, global_vars
+
+
+def gen_df_getitem_tuple_at_impl(self, row, col):
+    func_text, global_vars = df_getitem_tuple_at_codegen(self, row, col)
+    loc_vars = {}
+    exec(func_text, global_vars, loc_vars)
+    _reduce_impl = loc_vars['_df_getitem_tuple_at_impl']
+
+    return _reduce_impl
+
+
 @sdc_overload(operator.getitem)
 def sdc_pandas_dataframe_accessor_getitem(self, idx):
     if not isinstance(self, DataFrameGetitemAccessorType):
         return None
 
     accessor = self.accessor.literal_value
 
+    if accessor == 'at':
+        if isinstance(idx, types.Tuple) and isinstance(idx[1], types.Literal):
+            row = idx[0]
```

Comment:
Need to check that first item in idx is of the same type as index and second item is string